### ISSUE TITLE
Improved Rotations for Daggers

### ISSUE THREAD

**Issue by MsNyara** [b1ueb1ues/b1ueb1ues.github.io#25]
Hi! In the readme it says you're assuming 5C+FS combo for all daggers (since none has any exception added), I decided to do simple calculations with combo damage and SP charges with your frame data and I found that 4C+FS has 3.01%+ normal damage per frame and also 0.87% faster SP charges, which fits perfectly into Alex/Orion S1 costs as well.

4C+FS+4C+FS+1C, lasting 350 frames for 1113% modifier is also better than 5C+FS+5C lasting 390 frames for 1135% modifier by a total of 10.24% more normal damage per frame and 5.12% faster SP charge. More importantly, the first combo fits nearly perfectly into Francesca/Vice/Irfan/Renelle/Orsem/Luther/Renee/Melsa/Ezelith/Wedding Aoi S1, meanwhile the second has 144 SP leftover, just making bigger the SP charge difference in practice (11.42%+ in favor of the first combo then).

Vida, Fritz and Su Fang actually requires 4C+FS+4C+FS+2C to charge their S1, lasting 372 frames for 1264% modifier, for 16.75% more normal damage per frame for still 4.83% faster SP charges while at it. For Fleur and Dragonyule Cleo's S1 you need three 4C+FS rotations (492 frames, 1557% modifier), meanwhile it is 5C+FS+5C+FS+3C (503 frames, 1535% modifier), that is 2.2% faster charge and 3.70% damage per frame, though this considers last FS recovery being skipped by skill, otherwise there is no difference for them at least.

**S2/S3 Interaction:**

**4C+FS+4C+FS+1C Simple Rotation / 4C+FS+1C Half Rotation**
Francesca/Vice/Irfan/Renelle/Luther: Simple Rotation S1 x2 + C1 + S2 + Simple Rotation x1 + S3 + S1
Melsa: Simple Rotation S1 x2 + S2 + Simple Rotation S1 x1 + C1 + S3
Orsem: Simple Rotation S1 x2  + S2 + Simple Rotation x1 + S3 + S1
Renee: Simple Rotation S1 x2 + Half Rotation + S2 +Half Rotation + S3 + S1
Ezelith: Simple Rotation S1 x3 + C1 + S3 + Half Rotation + S1
Wedding Aoi: Simple Rotation S1 x2 + S2 + Simple Rotation S1 x1 + S3

**4C+FS+4C+FS Simple Rotation**
Alex/Orion: Simple Rotation S1 x2 + 3C + S2 + Simple Rotation S1 + C1 + S3

**4C+FS+4C+FS+C2 Simple Rotation**
Vida/Fritz: Simple Rotation S1 x2 + C2 + S2 + 4C+FS+4C + S3 + C2 + S1 (Skip C2 before S2 later)
Su Fang: Simple Rotation S1 x2 + Simple Rotation + S2 + S3 + S1

**4C+FS+4C+FS+4C+FS Simple Rotation**
Dragonyule Cleo: Simple Rotation S1 x1 + 4C+FS+3C +S2 + 4C+FS + Simple Rotation + S3 + S1
Fleur: No variations, just use S2 after S1 and do not mix S3 on it

All of those rotations surpass similar ones with 5C+FS except for Fritz and to some extent Dragonyule Cleo and Fleur.

**Comment by b1ueb1ues** [comment]
alex for example
( c5fs ) init
( c5s1 c5fsc5s1 c5s2 c5s1 c5s3 ) normal rotation
c5s1 c5fsc5s1 c5s2 c5s1 c5s3
c5s1 c5fsc5s1 c5s2 c5s1 c5s3
a normal rotation = 26.85s 
c5 * 6: 497 * 6 =  2982
fs * 1: 141
s1 * 3 + s2 + s3 = 3445.2
244.62 mod/s


( c4fsc4fss1 c4fsc4fss1 c3s2 c4fsc4fss1 c1s3 ) your rotation 
this rotation takes 29.5s
c4fs * 6 : 519 * 6 = 3114
c3: 259
c1: 75
s1 * 3 + s2 + s3 = 3445.2
233.67 mod/s

interrupt combo chain decrease dps badly .

I will check other sample later

**Comment by MsNyara** [comment]
https://docs.google.com/spreadsheets/d/1jYwb_gNKaHczJCF-svIm90pBP1G5dn3JzzPYM8gKBPY/edit?usp=sharing

Just browse to the two Alex pages. Alex case is kinda special due to the skill haste (which I forgot she had) and low S1 cost, otherwise it is not easily possible to chain so many C5>Skill combos in a row (and thus skip so many recoveries), which are indeed better than C4FS (or C4FS>Skill), however C4FS is better than C5FS in every case.

Another issue comes with Alex's insane conditional skill damage. C4FS chains usually involves a little more of normal damage into the mix, which decreases the gap between C5FS for Alex's case, with 1.3 mod instead 1.65, the difference grows larger in favor of C4FS combo chains (which also means non-conditional DPS improves even more for Alex's case).

Finally the current rotation involves a highly ugly inefficient but required starting combo, you can say it is just a single combo, but its impact is quite large for 60s simulation, and also lingers around even up to 180s simulation. C4FS combo chains does not involve any such inconvenience and is constant around 60s, 120s and 180s. This might or not be noticed depending of what was being executed by end point.

Basically Alex is one of the worst scenarios for C4FS, yet even there it fares positively over C5FS.

All this is also ignoring extra overdrive damage (thus more breaks/faster fights) and extra benefits for Francesca/Vida (maybe also Wedding Aoi, I gotta reserch that), it is also ignoring that C4FS is way easier to execute and allows to prevent more dodging through more constant FS repositioning.

**Comment by b1ueb1ues** [comment]
although 4cfs is more practical in real battle, but it's hard to find a 4cfs rotation to actually beat 5cfs in sandbag simulation.
melsa: 1760,  1719
![melsa](https://user-images.githubusercontent.com/13764293/61344483-fec4c000-a883-11e9-92f4-37dd0ec6c107.png)
orsem: 1856, 1664
![orsem](https://user-images.githubusercontent.com/13764293/61344485-01271a00-a884-11e9-8428-5f73ccf733eb.png)

**Comment by MsNyara** [comment]
https://docs.google.com/spreadsheets/d/1jYwb_gNKaHczJCF-svIm90pBP1G5dn3JzzPYM8gKBPY/edit?usp=sharing

I expanded it for Orsem and Melsa, though the gains for Melsa are smaller than I thought they would be, but still a gain anyway. Orsem's DPS might seem off due to different methodology regarding buffs (I'm not sure how you handle them), at least for my part I buff the modifiers implied in the following duration after activation, at least for self buffs with poor up time.

I tweaked the rotation a bit to include more recovery skips which was a short-sight of mine initially, likely required for all the rotations (I can get into that work for each case). Like in every case, there might be a gain if the simulation ends with a skill and a lose if the simulation ends during a SP charge cycle, I'm not sure how your simulation handles this peculiarity exactly, so my calculations are just a timeless average (at least for constant cycles).

**Comment by b1ueb1ues** [comment]
your melsa rotation works well
look forward to test others

**Comment by MsNyara** [comment]
I updated it to include a rotation for Fritz, his case is quite the special thing, though.

I changed Flash of Genius for Stellar Show since it affects his S2. Then I tweaked his rotation to make use of his bursting potential: he can use 2-3 boosted FS and his S1 during the S3 buff duration for massive damage!

His DPS jumped 200+ with just that, actually, however his cycle rotation (starting from green and ignoring orange) adds another 30 points as well, though the starting rotation is already almost one minute long, so it can only be appreciated in the 120s or 180s.

RR>FoG. SS>FP as well. Sisters' Day Out has a lot of potential behind it as it would allow for an extra burst window, but it would require a new starting rotation as well, I might dwell into that another day.

Hitting 8 knifes requires a big boss, though, S2 with 4 knifes scenario has 10.5% less DPS, though it is still superior to current conditions up scenario with some gap.

**Edit:** I made a small slip in the mod calculation for S2 (forgot to separate normal combo damage involved when interacting with FS mod), fixed, the result is still overwhelming positive for new rotation and still SS>FoG, but by a smaller margin now. Even without print change, the new rotation still beats by a generous margin the current one.

**Comment by MsNyara** [comment]
I included a Francesca SIM rotation page (the other two Francescas are for personal use), her rotation can be copied for Renelle and Vice without problem, it gives a modest boost for them.

Edit: I included individual pages for Renelle and Vice, it is the same rotation, though. I also made one for Irfan, which has a similar rotation, but it had to be adjusted for the S3 (namely chaining S1 and S2 on it).

**Comment by b1ueb1ues** [comment]
orsem can't s3 c1 s1 by hand.
even if you can do that by active auto, but the content need a meta rotation is more likely controled by hand or multiplayer. you have to notice the ui hiding 1.9s .

**Comment by b1ueb1ues** [comment]
bug?
![Screenshot from 2019-07-19 10-13-17](https://user-images.githubusercontent.com/13764293/61504446-f7312280-aa0d-11e9-87e6-0270b46c3c3c.png)

**Comment by MsNyara** [comment]
No, it is just a repeated sequence x3 times, it appears below, it would be too long to note those individually. Francesca FS is going to fare poorly on the simulation since her selling point is making quick breaks or guaranteeing a second break, also my two personal Francesca pages have the co-ability considered as well, it also has the plus that it is less affected by skill holding, I made a SIM page for her for your simulator.

About the skill chaining you're absolutely right, I forgot that point, I will review the rotations and adjust for that point.

**Comment by b1ueb1ues** [comment]
that's what x means , i c

**Comment by MsNyara** [comment]
x means the mod is multiplied by that amount, yeah, but I also use it to note other effects or note repetitions (which also multiply the frames/sp charge in that case, now using R symbol to avoid confusions). By the way I'm almost done adjusting the rotations for the skill chaining, I'm just finishing Fritz right now.

**Edit:** Done.

**Comment by b1ueb1ues** [comment]
she cannot end with c5- , it should be c5
need to move first c4fs into orange zone, and add an c4fs- at end. 
![Screenshot from 2019-07-19 11-53-32](https://user-images.githubusercontent.com/13764293/61508319-e38cb880-aa1b-11e9-9d41-eb24c2fdf9ce.png)

**Comment by b1ueb1ues** [comment]
and for those who don't have fs damage passive
i think c5-s1 is better than c4fs-s1
for example, move vice first c4fs to orange, and end with c5 c5- , or maybe end with c4fs c5- is better

**Comment by MsNyara** [comment]
Green combo replaces orange combo position for cycles after the starting rotation regardless their position in table, that is because it is kinda annoying to move stuff to another place due to how google doc tables copy/move works and I didn't know how each cycle would behave until finishing the rotation, same reason why Alex has some scrapped Empty's around.

I slipped in that I based SP charge solely on the S2 before and not in the C4FS at the start + the S2 before, to practical effect cycles 2+ starts as C4FS C5- S1.

Unchained C5 is pretty awful since it has a long recovery, even without FS mod the advantage for 4CFS is merely lasting less and packing more damage while at it. Chained it reverses since FS recovery is quite fast and does not benefit as much from animation cancel as C5.

**Comment by MsNyara** [comment]
**Some bug fixing:** 

For **_Orsem_** I made a small slip: one S1 didn't have enough charge, thus I had to swap **Q10** from C4- to C5- and **C17** from C2- to C1- to make up for it, there is almost no DPS change.

**_Francesca_**, **_Renelle_** and **_Vice_** **J10** (cycled only) had to be swapped from C1- to C2- to adjust for cycled SP costs for the S2, the loss is minimal (3-4 points). Irfan's **J10** (non-cycle) was swapped to C1- for the same reason, just inverse.

**_Alex's_** rotation had to be changed altogether from **J17** and forward for S3's SP costs, sadly. In the bright side I found **K10's** C5 would easily be changed into C4, so the loss is smaller at least. I also found a name label error: **H10** is C2, not C1.

**Comment by MsNyara** [comment]
**_Melsa's_** 60s simulation can use a small change: skip last C1 so her S1 fits in time.

**_Fritz's_** rotation was improved: **E17** was R3 C2FS, I changed it into C4FS, **F17** was C2- and was swapped to C4- to adjust for the new SP.

Added **_Luther_**, his rotation is nearly identical to Orsem aside **D10** has to be C3- instead C2- due to his S2 SP cost.

Added **_Su Fang_**, however... it might be possible C5FS was better for him after all, it just allowed his SP costs to cycle more smoothly. Anyway, feel free to try out!

Added **_Orion_**. His rotation starts at the blue point due to his skill prep with the skill prep SP in orange below and cycled in green. Swap orange with green for repetition cycles. If for any reason you ignore skill prep, I can make a new rotation based on Alex.

Added **_Vida_**. FoG>SS, FoG>FP and RR>FP since her S2 mod isn't high enough to make up the loss. I also tested if with this WP setup it was worth not using S2, however there is a very small DPS loss with that. Like Francesca, with SS she could use C2FS for more breaks and surpass RR (or FoG), but that is not considered in your simulator.

Added **_Renee_**. Her SP costs are cursed and never align (for 120s and 180s SIM), I had to end her rotation with a half-way S3 charged. If her rotation is repeated ignoring that fact she is going to lose 16 DPS in cycles based on a compensatory mod I made. I can over-extend her rotation up to 180s if necessary, at your call.

Added **_Fleur_**. We handle mods differently, but I'm still confident the new rotation will translate well in your simulator.

Added **_Ezelith_**. For 120s and 180s her SP costs are cursed and never align, I ended her rotation with her S2 63% charged. If her rotation is repeated ignoring that fact she is going to lose 48 DPS in cycles based in a compensatory mod I made. I can over-extend her rotation up to 180s if necessary, at your call. Maybe C5FS was better for her as well, at least I'm getting very awful numbers.

Added an **_Ezelith Debuff_**, this rotation is solely focused in her S2 debuff. It will likely also be inferior in your simulator, but I made it for fun! Maybe one day if they fix Flurry Debilitator this will be her definitive rotation. Even right now this should be her ideal rotation in raids, maybe worth adding for Special Page!

Added **_Wedding Aoi_**, nothing special, really, just sightly adjusted so her S1 is casted before her S2, in case you can hit sleep before the S2. Her off-conditional would require a different rotation, though.

Added **_Dragonyule Cleo_**, again, nothing special, aside delaying second S1 so it would fit inside her S3.

Reviewed all rotations for bug fixing SP costs/labels/etc. and no issues. Those are all~

**Comment by Ny-chan** [comment]
Made one for Ezelith, like Renee her SP costs are cursed and never align, irrelevant for 60s, but for 120s and 180s I just ended her rotation with a S3 which had more impact than her S2. I made a mod adjustment for that, however without mod the DPS for the cycles would be quite relevant over 40+, however a rotation with high S3 charge left can easily lose over 50+, so yeah. I can still over-extend her rotation all the way up to 180s for a precise simulation if necessary.

**Comment by MsNyara** [comment]
It seems your simulator does not consider the timing of buffs: it is not the same for a 5s buff to increase the average DPS over increasing the strongest proportion of it (a skill, usually), for example. I find reasonable to just buff the average DPS when it comes to random or semi-random factors, since the buff can hit at any given point, which includes teambuff, but otherwise it produces a bias toward longer lasting buffs. 

Meanwhile I'd strongly suggest including a timing mechanism for buffs in the simulator (for 5s and 10s at least), if that is not an option for any reason, I can make alternative rotations which would ignore buff timing for the following adventurers:

Irfan: Nearly identical to Francesca/Renelle/Vice.
Fritz: Nearly identical to Vida.
Orsem: Nearly identical to Melsa.
Luther: Nearly identical to Orsem.

Fleur, Renee, Su Fang, Dragonyule Cleo and Wedding Aoi would require more specific adjustments.

At your call.

**Comment by b1ueb1ues** [comment]
what are you talking about.
I do sim buff very accurate, that's why I found addis s2 buff so quickly, and sarisse benefit from 1 s3 2*s1.
And for the case you might not know, by manual control fleur s3 c3 s1 c3 s2(s3 end right after s2 here) c3 s1, it's not worth to wasting 1second for just cast s2 in s3 . So the best practise is s3 c5 s1 xx buffend x s2 xxx s1
 
![Screenshot from 2019-07-22 17-32-52](https://user-images.githubusercontent.com/13764293/61622224-f6ff8400-aca6-11e9-8ece-d5388bee8544.png)

**Comment by MsNyara** [comment]
Yeah, I know Fleur can't cast two S1 inside a S3, I found however that delaying S2 by doing C5- instead C3- before (as long there is enough SP already) negatively impacts her DPS, despite being 1.4 buffed.

**Comment by b1ueb1ues** [comment]
i notice you calculate s3 end after c3, but it's not true, if you do s3 s1 s2 s1, s3 end right after s2 cast.

**Comment by MsNyara** [comment]
Oh, I see, I guess my problem was not counting the 60f from S3 casting then. I guess I also got confused reading and trying to understand how your simulator works. In such case then there is no need to modify the rotations I made then.

**Comment by MsNyara** [comment]
Ok, after adjusting for that mistake (added the 60f from casting and now I'm getting the same end point) I noticed a couple changes for Fritz would do good, so I modified his rotation. Everyone else didn't need any modification (aside some mod adjustments I made for new end point for buff).

Su Fang and Ezelith (Normal) would use another rotation as well, though I can't figure well one to be honest, at least with C4FS charge. **Bug Fix:** Su Fang **B17** had to be C3- instead C2- due to the UI hiding, **P10** swapped C5- to C4-.

**Comment by MsNyara** [comment]
Ok, I noticed now why our Orsems DPS and other buff STR are differing so much. You are considering in-battle STR buffs additively to the whole STR %? Like 1.65 to 2.05. When I change my modifier from 1.4 into 1.24 (the DPS booat from 1.65 to 2.05 STR %) I get identical results with you, however... this is not how in-game works. I tried out how STR buffs stacks in-game with different adventurers and buffs, and it is always the % stated the increase (inside damage rolls posibilities) despite I have a 60% STR Dragon. The damage formula in gamepedia also agrees with my in-game test as it says in-battle buffs multiply all previous STR total instead additively stacking. Battle buffs only additively stacks with other battle buffs. It might also be cause of a bug somewhere.

**Comment by b1ueb1ues** [comment]
I am the contributer of the damage formula in gamepedia .
My buff additive to other buffs, but multiplicated with 60% str dragon.
buff is buff, passive is passive
my orsem c1 = 505, c1 with s3 = 707
707/505=1.4
and what's your 1.65 mean?

**Comment by MsNyara** [comment]
Dragonyule Jeanne 45% + 20% from Flash of Genius = 165% (1.65).

I get 706.95 damage from C1 40% boosted with the average crit rate/damage factored.
When I downgrade the boost to 24.24% (2.05/1.65), I get 627.37.
Without boost it is 504.97.

However nevermind, I forgot the current SIM rotation used the pre-bug fix Orsem still, when I changed back the rotation to pre-bug and I placed cut-point at 60s I got matching results (1863, 6 points higher, but that is likely due to decimal and rounding handling). Ironically the bug fix adjustment actually improved the DPS as well into 1872, for 60s cut point alone, though, since there wasn't enough saved time to start next combo, thus C5- C1- had a higher mod than C4- C2- before and not enough reduced duration for C4- C2- to surpass it with a new combo for 60s cut point. Since 1872 was relevantly higher than 1857 it had me wondering and I was looking out why.

Now I discovered the reason I will give a re-check and explore the rotations to see if any of them can be lightly modified for 60s simulation to adjust for cut point better (maybe for 120s too). Thanks you for your cooperation and sorry for the questions!
